tedana/utils/masking.py

```diff
@@ -0,0 +1,162 @@
+"""
+"""
+import numpy as np
+import pandas as pd
+import nibabel as nib
+
+
+def row_idx(arr1, arr2):
+    """
+    Get a 1D index of rows in arr1 that exist in arr2.
+
+    Parameters
+    ----------
+    arr1 : (X x 3) :obj:`numpy.ndarray`
+    arr2 : (Z x 3) :obj:`numpy.ndarray`
+
+    Returns
+    -------
+    idx : 1D :obj:`numpy.ndarray`
+        Index of rows in arr1 that exist in arr2.
+
+    Notes
+    -----
+    This works amazingly well, but is quite slow.
```

Comment:
We can make this much faster by subbing out `pandas`, if we stick with this approach !